src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,109 @@
+package mate.academy.dao;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exceptions.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.utils.ConnectorUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    @Override
+    public Book create(Book book) {
+        String sqlRequest = "INSERT INTO books(title, price) VALUES (?, ?)";
+        try (PreparedStatement statement = ConnectorUtil.getConnection().prepareStatement(
+                sqlRequest, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new RuntimeException("Expected at least 1 affected row, but was 0");
+            }
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                Long id = resultSet.getObject(1, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not create a book: " + book.toString());
```

Comment:
always add origin exception when you throw a new one
```suggestion
            throw new DataProcessingException("Can not create a book: " + book, e));
```